students/laud/session04/mailroom2.py

```diff
@@ -0,0 +1,223 @@
+#!/usr/bin/env python3
+import sys
+
+""" Historical donations data """
+donations = [
+    ("Bill Gates", [653772.32, 12.17]),
+    ("Jeff Bezos", [877.33]),
+    ("Paul Allen", [663.23, 43.87, 1.32]),
+    ("Mark Zuckerberg", [1663.23, 4300.87, 10432.00]),
+]
```

Comment:
In session 4 mailroom part 2, you are required to use a Python dictionary, not a list, so this needs to be changed.

In fact, browsing this source overall, this looks like the same as your session 3 mailroom part 1 source code. So it looks like you really haven't tried session 4 mailroom part 2 yet.

Please do try session 4 mailroom part 2 with a dictionary and file writing feature implemented.